src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1947 @@
+        under5_idx = df.index[df.is_alive & (df.age_exact_years < 5)]
+        # and ~df.un_am_treatment_type.isin(['standard_RUTF', 'soy_RUSF', 'CSB++', 'inpatient_care'])
+
+        def get_monitoring_frequency_days(age):
+            # TODO: maybe in future 0-1 to be dealt with within epi module
+            if age < 1:
+                return p['growth_monitoring_frequency_days_agecat'][0]
+            elif age <= 2:
+                return p['growth_monitoring_frequency_days_agecat'][1]
+            else:
+                return p['growth_monitoring_frequency_days_agecat'][2]
+
+        # schedule monitoring within age-dependent frequency
+        for person_id in under5_idx:
+            next_event_days = \
+                rng.randint(0, max(0, (get_monitoring_frequency_days(df.at[person_id, 'age_exact_years']) - 2)))
+            if (df.at[person_id, 'age_exact_years'] + (next_event_days / 365.25)) < 5:
+                self.sim.modules['HealthSystem'].schedule_hsi_event(
+                    hsi_event=HSI_Wasting_GrowthMonitoring(module=self.module, person_id=person_id),
+                    priority=2, topen=self.sim.date + pd.DateOffset(days=next_event_days)
+                )
+
+
+class HSI_Wasting_GrowthMonitoring(HSI_Event, IndividualScopeEventMixin):
+    """ Attendance is determined for the HSI. If the child attends, measurements with available equipment are performed
+    for that child. Based on these measurements, the child can be diagnosed as well/MAM/(un)complicated SAM and
+    eventually scheduled for the appropriate treatment. If the child (attending or not) is still under 5 at the time of
+    the next growth monitoring, the next event is scheduled with age-dependent frequency.
+    """
+
+    def __init__(self, module, person_id):
+        super().__init__(module, person_id=person_id)
+        assert isinstance(module, Wasting)
+
+        self.attendance = None
+
+        self.TREATMENT_ID = "Undernutrition_GrowthMonitoring"
+        self.ACCEPTED_FACILITY_LEVEL = '1a'
+
+    @property
+    def EXPECTED_APPT_FOOTPRINT(self):
+        """Return the expected appointment footprint based on attendance at the HSI event."""
+        rng = self.module.rng
+        p = self.module.parameters
+        person_age = self.sim.population.props.loc[self.target].age_exact_years
+
+        # TODO: for now assumed general attendance prob for <1 y old,
+        #  later may be excluded from here and be dealt with within epi module
+        def get_attendance_prob(age):
+            if age < 1:
+                return p['growth_monitoring_attendance_prob_agecat'][0]
+            if age <= 2:
+                return p['growth_monitoring_attendance_prob_agecat'][1]
+            else:
+                return p['growth_monitoring_attendance_prob_agecat'][2]
+
+        # perform growth monitoring if attending
+        self.attendance = rng.random_sample() < get_attendance_prob(person_age)
```

Comment:
this will mean if if no "attendance" there the HSI still runs, but just without any cost. That seems odd and probably unintended?
Better to do determine if attendance will happen when scheduling the "next appointment" and not scheduling one if it won't be attended.